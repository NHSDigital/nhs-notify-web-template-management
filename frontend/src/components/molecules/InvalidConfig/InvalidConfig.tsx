'use client';

import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';
import { InsetText } from 'nhsuk-react-components';
import content from '@content/content';
import { getBasePath } from '@utils/get-base-path';
import Link from 'next/link';

const { pageHeading, text, insetText, backLinkText } =
  content.pages.invalidConfiguration;

export default function InvalidConfig() {
  return (
    <NHSNotifyMain>
      <div className='nhsuk-grid-row' data-testid='page-content-wrapper'>
        <div className='nhsuk-grid-column-two-thirds'>
          <h1 className='nhsuk-heading-xl' data-testid='page-heading'>
            {pageHeading}
          </h1>
          <p>{text}</p>
          <InsetText visuallyHiddenText={false}>
            <p>{insetText}</p>
          </InsetText>
          <p>
            <Link
              href='/choose-a-template-type'
              data-testid='back-to-choose-template-type-link'
            >
              {backLinkText}
            </Link>
          </p>
        </div>
      </div>
    </NHSNotifyMain>
  );
}
