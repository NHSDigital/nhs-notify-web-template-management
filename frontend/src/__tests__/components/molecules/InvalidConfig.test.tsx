import InvalidConfig from '@molecules/InvalidConfig/InvalidConfig';
import { render } from '@testing-library/react';

test('InvalidConfig', async () => {
  const container = render(<InvalidConfig />);

  expect(container.asFragment()).toMatchSnapshot();
});
