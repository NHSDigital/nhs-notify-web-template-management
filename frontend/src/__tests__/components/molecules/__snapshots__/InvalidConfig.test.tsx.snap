// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidConfig 1`] = `
<DocumentFragment>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
      data-testid="page-content-wrapper"
    >
      <div
        class="nhsuk-grid-column-two-thirds"
      >
        <h1
          class="nhsuk-heading-xl"
          data-testid="page-heading"
        >
          You cannot create letter templates yet
        </h1>
        <p>
          To get access, contact your onboarding manager and give them this error message:
        </p>
        <div
          class="nhsuk-inset-text"
        >
          <p>
            Account needs a client ID and campaign ID
          </p>
        </div>
        <p>
          <a
            data-testid="back-to-choose-template-type-link"
            href="/templates/choose-a-template-type"
          >
            Go back
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
